File: backend/app/database/models/platform.py
import enum
from typing import Optional

from sqlmodel import Field, SQLModel, Relationship


class PlatformCategoryEnum(enum.Enum):
    conference_hall = "conference hall"
    restaurant = "restaurant"
    open_areas = "open areas"


class Platform(SQLModel, table=True):
    id: Optional[int] = Field(default=None, primary_key=True)
    name: str
    information: str
    address: str
    platform_category: PlatformCategoryEnum = Field(default="conference_hall")
    capacity: int
    price: int

    bookings: list["BookingInDB"] = Relationship(back_populates="platform")

File: frontend/composeApp/src/commonMain/kotlin/data/models/Platform.kt
package data.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable


@Serializable
data class Platform(
    val id: Int?,
    val name: String,
    val information: String,
    val address: String,
    @SerialName("platform_category")
    val platformCategory: String,
    val capacity: Int,
    val price: Int,
)